scripts/linters/pylint_extensions.py

```diff
@@ -2109,6 +2111,56 @@ def visit_call(self, node):
                     break
 
 
+class ConcatenationChecker(checkers.BaseChecker):
+    """Custom pylint checker which checks that string
+    concatenation is present or not
+    """
+
+    __implements__ = interfaces.ITokenChecker
+
+    name = 'string-concatenation'
+    priority = -1
+    msgs = {
+        'C0034': (
+            'at string %s, use string interpolation (\'string1%%s\' %% string2)'
+            ' rather than string concatenation (\'string1\' + \'string2\').',
+            'string-concatenation',
+            'Enforce use of string interpolation over string concatenation.',
+        ),
+    }
+
+    def process_tokens(self, tokens):
```

Comment:
Instead of the parsing tokens, can we do it through astroid/nodes? [For [ref](https://docs.google.com/document/d/1v0Oi2qFIC3HXwFaCX1zpG8aP0IjtZnYs_s_U8S-FHn8/edit?usp=sharing)]

I mean will it be possible to visit operators and if the operator is '+' and either of the left or right values are string then raise lint issue? (not sure!)

Reply:
**Done**

Reply:
Used astroid/nodes this time.